typescript/packages/wallets/zilliqa/src/ZilliqaWalletClient.ts

```diff
@@ -0,0 +1,106 @@
+//import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { Balance, Chain, Signature, type ToolBase, WalletClientBase } from "@goat-sdk/core";
+import {
+    type EVMReadRequest,
+    type EVMReadResult,
+    type EVMTransaction,
+    type EVMTypedData,
+    EVMWalletClient,
+} from "@goat-sdk/wallet-evm";
+import { ViemEVMWalletClient } from "@goat-sdk/wallet-viem";
+import { ViemOptions } from "@goat-sdk/wallet-viem";
+import { Account, Wallet } from "@zilliqa-js/account";
+import { Zilliqa } from "@zilliqa-js/zilliqa";
+import { type WalletClient as ViemWalletClient } from "viem";
+
+export abstract class ZilliqaWalletClient extends WalletClientBase {
+    abstract getZilliqaChainId(): number;
+
+    // This is as close as we're going to get to an abstract interface for the moment;
+    // when we add browser wallets, we will have to change this, but I'm reluctant
+    // to invent yet another Zilliqa-native transaction abstraction before we have to.
+    abstract getZilliqa(): Zilliqa;
+
+    abstract getEVM(): EVMWalletClient;
+}
+
+// Zilliqa has two APIs - EVM and native. The native API is provided by the Zilliqa object. For the EVM API,
+// we thunk down to viem.
+//
+// We need to encapsulate, rather than extend ViemEVMWalletClient so that we can report our chain type correctly.
+// One day it would be nice to add the ability to use ZilPay client integration, but those libraries don't exist yet,
+// so we do this instead.
+export class ZilliqaJSViemWalletClient extends ZilliqaWalletClient {
+    zilliqa: Zilliqa;
+    viem: ViemEVMWalletClient;
+    chainId = 0;
+
+    constructor(client: ViemWalletClient, node: string, account: Account, chainId: number, options?: ViemOptions) {
+        super();
+        this.viem = new ViemEVMWalletClient(client, options);
+        this.zilliqa = new Zilliqa(node);
+        this.zilliqa.wallet.addByPrivateKey(account.privateKey);
+        this.chainId = chainId;
+    }
+
+    getZilliqaChainId(): number {
+        return this.chainId;
+    }
+
+    // Use the EVM address here - callers expecting the Zilliqa address will hopefully know
+    // to get an address from getZilliqa() instead.
+    getAddress(): string {
+        return this.viem.getAddress();
+    }
+
+    getCoreTools(): ToolBase[] {
```

Comment:
No need to add this, `WalletClientBase` already has this :)

Reply:
Good point well made - thanks - removed.